### PR TITLE
Fix long call stack in downloader

### PR DESCRIPTION
**Pull request by pandamicro**
Re: cocos-creator/fireball#5816

Changes proposed in this pull request:
 * Fix long call stack in downloader

> Makes sure these boxes are checked before submitting your PR - thank you!
>
- [x] If your pull request has gone "stale", you should **rebase** your work on top of the latest version of the upstream branch.
- [x] If your commit history is full of small, unimportant commits (such as "fix pep8" or "update tests"), **squash** your commits down to a few, or one, discreet changesets before submitting a pull request.
- To official teams:
  - [x] Check that your javascript is following our [style guide](https://github.com/cocos-creator/fireball/blob/dev/.github/CONTRIBUTING.md) and end files with a newline
  - [x] Document new code with comments in source code based on [API Docs](https://github.com/cocos-creator/fireball#api-docs)
  - [x] Make sure any runtime log information in `cc.log` , `cc.error`, `cc.warn` or `cc.assert` has been moved into `DebugInfos.js` with an ID

@cocos-creator/engine-admins

### CODE REVIEW

**Review by jareguo — APPROVED** [review]

File: cocos2d/core/platform/CCMacro.js [hunk 1]
```diff
@@ -1898,6 +1898,16 @@ cc.macro = {
...
+
+    /**
+     * !#en 
+     * The max concurrent task number for the downloader
+     * !#zh
+     * 下载任务的最大并发数限制，在安卓平台部分机型或版本上可能需要限制在较低的水平
+     * @property {Number} DOWNLOAD_MAX_CONCURRENT
+     * @default 64
+     */
+    DOWNLOAD_MAX_CONCURRENT: 64,
```
Comment (jareguo) [inline cocos2d/core/platform/CCMacro.js:1910]: 我刚刚查阅了某鹭的最新代码，他的最大并发数仍然是 2，建议 1.5.x 这里还是先不动了，或者区分下 JSB 与否

Comment (pandamicro) [reply cocos2d/core/platform/CCMacro.js:1910]: 我建议是让燕飞先测一下，看看现在的情况是什么样，会不会在一些机型上加载变很慢，比如用 DarkSlash / BlackJack 这种资源比较多的项目来测试。浏览器和操作系统一直在更新换代，一两年以前的经验在今天不一定适用。

最后可以保守一点，我们在 main.js 里面添加判断，如果是 Android 则限制为 2，其他平台不限制，这样用户也可以自由去根据自己的测试情况去设置

Comment (jareguo) [reply cocos2d/core/platform/CCMacro.js:1910]: 那你单独发一个可配置的版本让他测试吧，这里测完再合并

Comment (jareguo) [reply cocos2d/core/platform/CCMacro.js:1910]: 测试项目我单独发给他

Comment (pandamicro) [reply cocos2d/core/platform/CCMacro.js:1910]: 添加到 main.js 了，不需要修改这个 PR

**Pull request merged**